spring-cloud-dataflow-docs/src/main/asciidoc/appendix-migration-guide.adoc

```diff
@@ -0,0 +1,381 @@
+[appendix]
+[[migration-guide]]
+== Migrating from Spring XD to Spring Cloud Data Flow
+
+[partintro]
+--
+In this section you will learn all about the migration path from Spring XD to Spring Cloud Data Flow 
+along with the tips and tricks.
+--
+
+=== Terminology Changes
+
+[width="100%",frame="topbot",options="header"]
+|======================
+|Old |New
+|XD-Admin        |Server (_implementations_: local, cloud foundry, apache yarn, kubernetes, and apache mesos)
+|XD-Container       |N/A
+|Modules       |Applications
+|Admin UI        |Dashboard
+|Message Bus        |Binders
+|Batch / Job        |Task
+|======================
+
+=== Modules to Applications
+If you have custom Spring XD modules, you’d have to refactor them to use Spring Cloud 
+Stream and Spring Cloud Task binding annotations, with updated dependencies and built as normal 
+Spring Boot "applications". 
+
+==== Custom Applications
+
+* Spring XD's stream and batch modules are refactored into link:https://github.com/spring-cloud/spring-cloud-stream-app-starters[Spring Cloud Stream] and link:https://github.com/spring-cloud/spring-cloud-task-app-starters[Spring 
+Cloud Task] application-starters, respectively. These applications can be used as the reference while refactoring Spring XD modules
+* There are also some samples for link:https://github.com/spring-cloud/spring-cloud-stream-samples[Stream] and link:https://github.com/spring-cloud/spring-cloud-task/tree/master/spring-cloud-task-samples[Task] applications for reference
+* If you’d like to create a brand new custom application, use the getting started guide for link:https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc#getting-started[Stream] and link:https://github.com/spring-cloud/spring-cloud-task/blob/master/spring-cloud-task-docs/src/main/asciidoc/getting-started.adoc#developing-your-first-spring-cloud-task-application[Task] applications and as well as  review the development link:https://github.com/spring-cloud/spring-cloud-stream-app-starters/blob/master/spring-cloud-stream-app-starters-docs/src/main/asciidoc/overview.adoc#creating-your-own-applications[guide]
+* Alternatively, if you’d like to patch any of the out-of-the-box stream/task applications, you can 
```

Comment:
On this section just state stream application.  Because the dependencies enumerated in the link are for streams only.